scripts/generate-changelog.sh

```
@@ -0,0 +1,222 @@
+    categories[chore]=""
+    categories[style]=""
+    categories[revert]=""
+    
+    # Process each commit
+    while IFS= read -r commit; do
+        if [[ -z "$commit" ]]; then
+            continue
+        fi
+        
+        # Extract type and description
+        if [[ $commit =~ ^([a-z]+)(\([^)]+\))?:\ (.+)$ ]]; then
```

Shellcheck (0.10.0)

[error] Couldn't parse this test expression. Fix to allow more checks.

(SC1073)